BurnOutSharp/ProtectionType/SafeDisc.cs

```diff
@@ -151,6 +330,236 @@ public static string GetVersion(string file, byte[] fileContent, List<int> posit
             return $"{version}.{subVersion:00}.{subsubVersion:000}";
         }
 
+        public static string Get00000001TMPVersion(string firstMatchedString, IEnumerable<string> files)
+        {
+            if (firstMatchedString == null || !File.Exists(firstMatchedString))
```

Comment:
For all of the places that you have `firstMatchedString == null` can you replace it with `string.IsNullOrEmpty(firstMatchedString)`? It helps capture more issues.